package/tests/test_integrations/test_hooks_utils.py

```diff
@@ -0,0 +1,255 @@
+
+    class MockDataset:
+        def __init__(self, filepath=None, _filepath=None):
+            if filepath:
+                self.filepath = filepath
+            if _filepath:
+                self._filepath = _filepath
+
+    return {
+        "dataset_with_filepath": MockDataset(filepath="/path/to/file.csv"),
+        "dataset_with_private_filepath": MockDataset(
+            _filepath="/private/path/file.csv"
+        ),
+        "dataset_without_filepath": MockDataset(),
+    }
+
+
+class TestHashNode:
+    """Tests for hash_node function."""
+
+    def test_hash_node_with_kedro_node(self, example_kedro_node, mocker):
+        """Test hash_node function with KedroNode."""
+        mock_hash = mocker.patch(
+            "kedro_viz.integrations.kedro.hooks_utils._hash", return_value="test_hash"
+        )
+
+        result = hash_node(example_kedro_node)
+
+        assert result == "test_hash"
+        mock_hash.assert_called_once_with(str(example_kedro_node))
+
+    def test_hash_node_with_string(self, mocker):
+        """Test hash_node function with string input."""
+        mock_hash_input_output = mocker.patch(
+            "kedro_viz.integrations.kedro.hooks_utils._hash_input_output",
+            return_value="test_io_hash",
+        )
+
+        test_input = "test_dataset"
+        result = hash_node(test_input)
+
+        assert result == "test_io_hash"
+        mock_hash_input_output.assert_called_once_with(test_input)
+
+
+class TestCreateDatasetEvent:
+    """Tests for create_dataset_event function."""
+
+    def test_create_dataset_event_basic(self):
+        """Test create_dataset_event function with basic parameters."""
+        event_type = "after_dataset_loaded"
+        dataset_name = "test_dataset"
+
+        result = create_dataset_event(event_type, dataset_name)
+
+        expected = {
+            "event": event_type,
+            "dataset": dataset_name,
+            "node_id": hash_node(dataset_name),
+            "status": "Available",
+        }
+
+        assert result == expected
+
+    def test_create_dataset_event_with_size(self, mock_datasets, mocker):
+        """Test create_dataset_event function with size calculation."""
+        mock_compute_size = mocker.patch(
+            "kedro_viz.integrations.kedro.hooks_utils.compute_size", return_value=1024
+        )
+
+        event_type = "after_dataset_saved"
+        dataset_name = "test_dataset"
+        dataset_value = "test_data"
+
+        result = create_dataset_event(
+            event_type, dataset_name, dataset_value, mock_datasets
+        )
+
+        expected = {
+            "event": event_type,
+            "dataset": dataset_name,
+            "node_id": hash_node(dataset_name),
+            "status": "Available",
+            "size_bytes": 1024,
+        }
+
+        assert result == expected
+        mock_compute_size.assert_called_once_with(
+            dataset_name, dataset_value, mock_datasets
+        )
+
+
+class TestComputeSize:
```

Comment:
Why do you need so much mocking for this method? Util functions should usually not need mocking to be tested.